.github/workflows/deploy_staging.yml

```diff
@@ -31,25 +31,79 @@ jobs:
 
       - name: Decrypt large secret
         run: ./.github/workflows/decrypt_secret.sh
         env:
           LARGE_SECRET_PASSPHRASE: ${{ secrets.LARGE_SECRET_PASSPHRASE }}
 
-      # Bundle requires FAB connected to Play Store to upload https://appdistribution.page.link/KPoa
+      # Build Android
+      # Bundle requires Firebase connected to Play Store to upload https://appdistribution.page.link/KPoa
       - name: Build Android apk
         run: flutter build apk --flavor staging --debug --build-number $GITHUB_RUN_NUMBER
 
-      # Upload artifact to Staging Firebase App Distribution
-      - name: Upload Staging to Firebase
+      - name: Upload apk artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: app-staging-debug.apk
+          path: build/app/outputs/flutter-apk/app-staging-debug.apk
+          if-no-files-found: error
+
+      # Build iOS
+      - name: Install distribution certificate & provisioning profile
+        run: |
+          cd ios
+          fastlane codesigning keychain_password:${{ secrets.IOS_KEYCHAIN_PASSPHASE }} certificate_password:${{ secrets.IOS_PRIVATE_KEY_P12_PWD }}
+
+      - name: Pod install
+        run: |
+          cd ios
+          pod install
+
+      - name: Build iOS archive
+        run: flutter build ipa --flavor staging --release --build-number $GITHUB_RUN_NUMBER --export-options-plist=ios/config/ExportOptions.plist
+
+      - name: Upload ipa artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: flutter_survey.ipa
+          path: build/ios/ipa/flutter_survey.ipa
+          if-no-files-found: error
+
+  deploy_on_linux:
+    name: Deploy builds into Firebase
+    needs: build_on_mac
+    runs-on: ubuntu-latest
+    environment: staging
+    steps:
+      - name: Download apk artifact
+        uses: actions/download-artifact@v2
+        with:
+          name: app-staging-debug.apk
+
+      # This container action is only supported on Linux https://github.com/wzieba/Firebase-Distribution-Github-Action/issues/24
+      - name: Upload Android Staging to Firebase
         uses: wzieba/Firebase-Distribution-Github-Action@v1
         with:
           appId: ${{secrets.FIREBASE_ANDROID_APP_ID_STAGING}}
           token: ${{secrets.FIREBASE_TOKEN}}
           groups: dev
-          file: build/app/outputs/flutter-apk/app-staging-debug.apk
+          file: app-staging-debug.apk
+
+      - name: Download ipa artifact
+        uses: actions/download-artifact@v2
+        with:
+          name: flutter_survey.ipa
+
+      # This container action is only supported on Linux https://github.com/wzieba/Firebase-Distribution-Github-Action/issues/24
+      - name: Upload iOS Staging to Firebase
+        uses: wzieba/Firebase-Distribution-Github-Action@v1
+        with:
+          appId: ${{secrets.FIREBASE_IOS_APP_ID_STAGING}}
+          token: ${{secrets.FIREBASE_TOKEN}}
+          groups: dev
+          file: flutter_survey.ipa
```

Comment:
@luongvo I think we can separate them into 2 jobs to upload to Firebase parallelly

Reply:
@manh-t `Firebase-Distribution-Github-Action` does not work on macOS host, that means we need at least 3 jobs, right? 🤔

Reply:
@luongvo yeah, we may need 3 jobs, the first job will build and upload Android and iOS build to artifact, the rest 2 jobs will be executed after the first job's done. So when the first job gets an error, the later jobs will not be executed 🤔 . That's my idea 😁

Reply:
> @luongvo yeah, we may need 3 jobs, the first job will build and upload Android and iOS build to artifact, the rest 2 jobs will be executed after the first job's done. So when the first job gets an error, the later jobs will not be executed 🤔 . That's my idea 😁

if we want to fail them in case of error, why do we need to separate the jobs?

Reply:
@luongvo the 2 jobs will execute parallelly, and it saves time for us rather than perform uploading Android then iOS sequentially, doesn't it?

Reply:
@manh-t ok let me try then

Reply:
@manh-t improved by 3 jobs now. I decided to separate build tasks between Android and iOS, even 1 build failed, we still continue with another ✅  https://github.com/luongvo/flutter-survey/runs/3552500655?check_suite_focus=true